code/game/objects/items/weapons/ecigs.dm

```diff
@@ -0,0 +1,296 @@
+	. = ..()
+	color = pick(ecig_colors)
+
+obj/item/clothing/mask/smokable/ecig/util/examine(mob/user)
+	. = ..()
+	if(ec_cartridge)
+		to_chat(user, SPAN_NOTICE("There are [round(ec_cartridge.reagents.total_volume, 1)] units of liquid remaining."))
+	else
+		to_chat(user, SPAN_NOTICE("There's no cartridge connected."))
+	if(cig_cell)
+		to_chat(user, SPAN_NOTICE("The power meter shows that there's about [round(cig_cell.percent(), 5)]% power remaining."))
+	else
+		to_chat(user, SPAN_NOTICE("There's no cartridge connected."))
+	if(active)
+		to_chat(user, SPAN_NOTICE("It is currently turned on."))
+	else
+		to_chat(user, SPAN_NOTICE("It is currently turned off."))
+
+/obj/item/clothing/mask/smokable/ecig/deluxe
+	name = "deluxe electronic cigarette"
+	desc = "A premium model eGavana MK3 electronic cigarette, shaped like a cigar."
+	icon_state = "pcigoff1"
+	icon_off = "pcigoff1"
+	icon_empty = "pcigoff2"
+	icon_on = "pcigon"
+	cell_type = /obj/item/cell/crap //enough for five catridges
+
+/obj/item/clothing/mask/smokable/ecig/deluxe/examine(mob/user)
+	. = ..()
+	if(ec_cartridge)
+		to_chat(user, SPAN_NOTICE("There are [round(ec_cartridge.reagents.total_volume, 1)] units of liquid remaining."))
+	else
+		to_chat(user, SPAN_NOTICE("There's no cartridge connected."))
+	if(cig_cell)
+		to_chat(user, SPAN_NOTICE("The power meter shows that there's about [round(cig_cell.percent(), 1)]% power remaining."))
+	else
+		to_chat(user, SPAN_NOTICE("There's no cartridge connected."))
+
+/obj/item/clothing/mask/smokable/ecig/proc/deactivate()
+	active = FALSE
+	STOP_PROCESSING(SSprocessing, src)
+	update_icon()
+
+/obj/item/clothing/mask/smokable/ecig/process()
+	if(!cig_cell)
+		deactivate()
+		return
+	if(!ec_cartridge)
+		deactivate()
+		return
+
+	if(idle >= idle_treshold) //idle too long -> automatic shut down
+		idle = 0
+		visible_message(SPAN_NOTICE("\The [src] powers down automatically."), null, 2)
+		deactivate()
+		return
+
+	idle ++
+
+	if(ishuman(loc))
+		var/mob/living/carbon/human/C = loc
+
+		if (!active || !ec_cartridge || !ec_cartridge.reagents.total_volume)//no cartridge
+			if(!ec_cartridge.reagents.total_volume)
+				to_chat(C, SPAN_NOTICE("There's no liquid left in \the [src], so you shut it down."))
+			deactivate()
+			return
+
+		if (src == C.wear_mask && C.check_has_mouth()) //transfer, but only when not disabled
+			idle = 0
+			//here we'll reduce battery by usage, and check powerlevel - you only use batery while smoking
+			if(!cig_cell.checked_use(power_usage * CELLRATE)) //if this passes, there's not enough power in the battery
+				deactivate()
+				to_chat(C,SPAN_NOTICE("\The [src]'s power meter flashes a low battery warning and shuts down."))
+				return
+			ec_cartridge.reagents.trans_to_mob(C, REM, CHEM_BREATHE, 0.4) // Most of it is not inhaled... balance reasons.
+
+/obj/item/clothing/mask/smokable/ecig/update_icon()
+	if (active)
+		item_state = icon_on
+		icon_state = icon_on
+		set_light(0.6, 0.5, brightness_on)
```

Comment:
i don't think our lighting system can use values that low, check the minimum values inside the proc. i think range is a minimum of 1.4